src/frontend/src/handler/show.rs

```diff
@@ -131,6 +133,74 @@ pub fn handle_show_object(handler_args: HandlerArgs, command: ShowObject) -> Res
     ))
 }
 
+pub fn handle_show_create_object(
+    handle_args: HandlerArgs,
+    show_create_type: ShowCreateType,
+    name: ObjectName,
+) -> Result<RwPgResponse> {
+    let session = handle_args.session;
+    let catalog_reader = session.env().catalog_reader().read_guard();
+    let (schema_name, object_name) =
+        Binder::resolve_schema_qualified_name(session.database(), name.clone())?;
+    let schema_name = schema_name.unwrap_or(DEFAULT_SCHEMA_NAME.to_string());
+    let schema = catalog_reader.get_schema_by_name(session.database(), &schema_name)?;
+    let sql = match show_create_type {
+        ShowCreateType::MaterializedView => {
+            let table = schema.get_table_by_name(&object_name).ok_or_else(|| {
+                RwError::from(CatalogError::NotFound(
+                    "materialized view",
+                    name.to_string(),
+                ))
+            })?;
+            if !table.is_mview() {
+                return Err(CatalogError::NotFound("materialized view", name.to_string()).into());
+            }
+            // We only stored the definition of materialized view, format and return directly.
+            format!(
+                "CREATE MATERIALIZED VIEW {} AS {}",
+                table.name, table.definition
+            )
+        }
+        ShowCreateType::View => {
+            let view = schema
+                .get_view_by_name(&object_name)
+                .ok_or_else(|| RwError::from(CatalogError::NotFound("view", name.to_string())))?;
+
+            // We only stored original sql in catalog, uses parser to parse and format.
+            let stmt = Parser::parse_sql(&view.sql)
+                .map_err(|err| anyhow!("Failed to parse view create sql: {}, {}", view.sql, err))?;
+            assert!(stmt.len() == 1);
+            stmt[0].to_string()
```

Comment:
The SQL unparsed from the AST might be incorrect. :( https://github.com/risingwavelabs/risingwave/issues/6801 Will this be a problem, if this is for some programs to read?

Reply:
If so, I guess we need to carefully review the `fmt::Display` for all AST nodes and add a comprehensive test suite for it.

Reply:
🥵 I think this is just for user to check how the mview and view are created.

> If so, I guess we need to carefully review the `fmt::Display` for all AST nodes and add a comprehensive test suite for it.

I think this's also needed, does the tests in sql_parser meet demand? It seems that we only have few tests there.

Reply:
I'm not sure. cc @lmatz

Reply:
We can test roundtrip parsing (`parse(unparse(parse(sql))) == parse(sql)` ) in sqlsmith

Reply:
Agree.

Reply:
> We can test roundtrip parsing (`parse(unparse(parse(sql))) == parse(sql)` ) in sqlsmith.

I'm afraid there's a case that some different SQLs are parsed to the same AST, which requires some binder context to distinguish. So when we unparse it, we cannot find the original representation, and this roundtrip test cannot cover this.

Reply:
I guess that's exactly why using AST instead of original sql to compare in xxchan's proposal?